Sources/SwiftFormat/Rules/UseShorthandTypeNames.swift

```diff
@@ -47,24 +47,28 @@ public final class UseShorthandTypeNames: SyntaxFormatRule {
 
     switch node.name.text {
     case "Array":
-      guard let typeArgument = genericArgumentList.firstAndOnly else {
+      guard let argument = genericArgumentList.firstAndOnly,
+            case .type(let typeArgument) = argument else {
```

Comment:
I think you can fuse this into 

```suggestion
      guard case .type(let typeArgument) = genericArgumentList.firstAndOnly {
```

Same comment in most matching cases below